arangod/Aql/IndexNodeOptimizerRules.cpp

```diff
@@ -45,13 +45,13 @@ namespace {
         }
         auto indexId = index->id();
         // use one index only
-        if (commonIndexId != 0 && commonIndexId != indexId) {
+        if (commonIndexId != std::numeric_limits<TRI_idx_iid_t>::max() && commonIndexId != indexId) {
           continue;
         }
         size_t indexFieldNum = 0;
         for (auto const& field : index->fields()) {
           if (arangodb::basics::AttributeName::isIdentical(nodeAttr.attr, field, false)) {
-            if (commonIndexId == 0) {
+            if (std::numeric_limits<TRI_idx_iid_t>::max() == commonIndexId) {
```

Comment:
Could you switch order for consistentcy
 commonIndexId == std::numeric_limits<TRI_idx_iid_t>::max()

Reply:
I used https://en.wikipedia.org/wiki/Yoda_conditions

Reply:
It will be as commonIndexId.isNone() in devel later.